src/main.rs

```diff
@@ -36,6 +48,93 @@ fn save_image(image: Vec<u8>, name: &str) -> Result<(), io::Error> {
+    Command::new("python")
+            .arg(MODEL_SERVER_PATH)
+            .arg(model_path)
+            .arg(format!("{}", MODEL_SERVER_PORT))
+            .spawn()
+}
+
+fn kill_model(mut model_proc: Child) -> Result<(), io::Error> {
+    model_proc.kill()
+}
+
+fn handle_request(req: Request) -> Result<Response, io::Error> {
+    // Spawn model proceess to send request to.
+    let model_proc = match req.expression {
+        Expression::Anger => {
+            match start_model(ANGER_MODEL_PATH) {
+                Ok(proc) => proc,
+                Err(e) => {
+                    error!("failed to start anger model: {}", e);
+                    return Err(e);
+                },
+            }
+        },
+        Expression::Happiness => {
+            match start_model(HAPPINESS_MODEL_PATH) {
+                Ok(proc) => proc,
+                Err(e) => {
+                    error!("failed to start happiness model: {}", e);
+                    return Err(e);
+                },
+            }
+        },
+    };
+
+    // Save the image to be processed by the model server.
+    if let Err(e) = save_image(req.image, IMG_NAME) {
+        error!("save image failed: {}", e);
+        kill_model(model_proc).unwrap();
+        return Err(e);
+    }
+
+    // Sleep to ensure model process is ready.
+    thread::sleep(Duration::from_millis(6000));
+
+    // Send prediction request to child proc and listen for result.
+    let prediction = match TcpStream::connect(format!("127.0.0.1:{}", MODEL_SERVER_PORT)) {
+        Ok(mut stream) => {
+            let mut cwd = env::current_dir().unwrap();
+            cwd.push(IMG_NAME);
+            let img_path = String::from(cwd.to_str().unwrap());
+            if let Err(e) = stream.write(img_path.as_bytes()) {
+                error!("failed to writing request to model server: {}", e);
+                kill_model(model_proc).unwrap();
+                return Err(e);
+            }
+            
+            let mut buffer = [0 as u8; BUFFER_SIZE];
+            match stream.read(&mut buffer) {
+                Ok(_) => {
+                    let pred_str = String::from_utf8(vec![buffer.to_vec()[0]]).unwrap();
+                    pred_str.trim().parse::<u8>().unwrap()
+                },
+                Err(e) => {
+                    error!("failed reading from model server: {}", e);
+                    kill_model(model_proc).unwrap();
+                    return Err(e);
+                },
+            }
+        },
+        Err(e) => {
+            error!("failed to connect to model server: {}", e);
+            kill_model(model_proc).unwrap();
+            return Err(e);
+        }
+    };
+
+    // All done, kill the model server.
+    kill_model(model_proc).unwrap();
+
+    // Create and return prediction response.
+    match prediction {
+        1 => Ok(Response::Accept{matches_expression: true}),
+        _ => Ok(Response::Accept{matches_expression: false}),
```

Comment:
Does the model return 0 when the image is not a match? Check for 0 here and print an error if the value is anything else.